org.openhab.binding.simatic/src/main/java/org/openhab/binding/simatic/internal/SimaticTCP.java

```diff
@@ -326,26 +337,33 @@ public void checkNewData() {
 
             if (!readLock.tryLock()) {
                 if (logger.isDebugEnabled()) {
-                    logger.debug("{} - Reading allready in progress", toString());
+                    logger.debug("{} - Reading already in progress", toString());
                 }
                 return;
-            }
 
-            readLock.lock();
+            }
 
+            logger.debug("Locking");
+            // tryLock() has already acquired the lock, this is not needed -- AchilleGR
+            // readLock.lock();
             for (SimaticReadDataArea area : readAreasList.getData()) {
                 byte[] buffer = new byte[area.getAddressSpaceLength()];
 
                 int result;
                 try {
+                    // If the connection is closed, this is null -- AchilleGR
+                    if (dc == null) {
```

Comment:
It used to happen when I was calling item.getValue().close() in SimaticBinding.java. I commented out this call later on and let your reconnect code take care of reopening the connection, so this is probably not hapenning anymore. It doesn't hurt to check though just in case.